### PR TITLE
[BUGFIX] Remove vestigial path normalization in chains

### PR DESCRIPTION
**Pull request by mmun**
This fixes a bug that occurs when chaining through a property that starts with a capital letter.

[Here is a JSBin](http://emberjs.jsbin.com/veromo/1/edit?html,js,output) with a demonstration of the bug.

### CODE REVIEW

File: packages/ember-metal/lib/binding.js [hunk 1]
```diff
@@ -169,19 +162,45 @@ Binding.prototype = {
...
+    // whether the path refers to a global or not by consulting `Ember.lookup`.
+    if (isGlobalPath(this._from)) {
+      let name = getFirstKey(this._from);
+      let possibleGlobal = Ember.lookup[name];
+
+      if (possibleGlobal) {
+        deprecate(
+          `Connecting an Ember.Binding to a global path such as '${this._from}' is deprecated. Please use a service instead.`,
+          false,
+          { id: 'ember-metal.global-bindings', until: '3.0.0' }
```
Comment (mixonic) [inline packages/ember-metal/lib/binding.js]: Can you please make an entry in the [deprecation guide](https://github.com/emberjs/website/blob/master/source/deprecations/v2.x.html.md) and add a `url` property here.

Comment (mmun) [inline packages/ember-metal/lib/binding.js]: Let's discuss whether we want to deprecate all of Ember.Binding first.

Comment (mmun) [inline packages/ember-metal/lib/binding.js]: (On that note I am going to remove the deprecation from this PR)

**Comment by mmun** [comment]
Kris pointed out a couple other things that can be cleaned up:

- https://github.com/emberjs/ember.js/blob/59fa9721c2278d2d23d4d4417b00ff726e60167d/packages/ember-metal/lib/chains.js#L153
- https://github.com/emberjs/ember.js/blob/59fa9721c2278d2d23d4d4417b00ff726e60167d/packages/ember-metal/lib/chains.js#L335

And we should do some performance testing to see how much of a win it is.

**Comment by stefanpenner** [comment]
:fire:

**Comment by rwjblue** [comment]
Thanks @mmun!

File: packages/ember-metal/lib/binding.js [hunk 2]
```diff
@@ -219,20 +232,20 @@ Binding.prototype = {
 
   /* Called when the from side changes. */
   fromDidChange(target) {
```
Comment (stefanpenner) [inline packages/ember-metal/lib/binding.js]: this arg doesn't seem one used anymore?

**Comment by krisselden** [comment]
This should be documented as breaking but since the behavior made no sense and didn't align with templates I doubt this will bite anyone

**Pull request merged**